docs/datatypes/tref.md

```diff
@@ -0,0 +1,223 @@
+---
+id: datatypes_tref
+title:  "TRef"
+---
+
+A `TRef[A]` is part of the `STM` bundle that represents a transactional reference that can perform reads and writes inside a transactional effect.
+
+`TRef` gives the basic machinery for others `SMT` transactional data types.
+
+`TRef[A]` where `A` is an immutable value.
+
+## Create a TRef
+
+Creating a `TRef` without committing its value:
+
+```scala mdoc:silent
+import zio.stm.{STM, TRef}
+
+val tVar: STM[Nothing, TRef[Int]] = TRef.make(10)
+```
+
+Or creating a `TRef` and immediately committing its value to extract out the value:
+
+```scala mdoc:silent
+import zio.UIO
+import zio.stm.TRef
+
+val tVar: UIO[TRef[Int]] = TRef.makeCommit(10)
+```
+
+## Retrieve the value out of a TRef
+
+By retrieving the value in a single transactional statement: 
+
+```scala mdoc:silent
+import zio.UIO
+import zio.stm.TRef
+
+val tVal: UIO[Int] = (for {
+  tRef <- TRef.make(10)
```

Comment:
See the comment about renaming.

Reply:
Sorry, I didn't pull the latest changes. Thanks for noticing. +1

Reply:
As you made changes to rolling back to `make`, I will leave as it is.